crates/cdk-lnd/src/lib.rs

```diff
@@ -204,47 +205,146 @@ impl MintLightning for Lnd {
             }
         };
 
-        let pay_req = fedimint_tonic_lnd::lnrpc::SendRequest {
-            payment_request,
-            fee_limit: max_fee.map(|f| {
-                let limit = Limit::Fixed(u64::from(f) as i64);
-
-                FeeLimit { limit: Some(limit) }
-            }),
-            amt_msat: amount_msat as i64,
-            ..Default::default()
-        };
-
-        let payment_response = self
-            .client
-            .lock()
-            .await
-            .lightning()
-            .send_payment_sync(fedimint_tonic_lnd::tonic::Request::new(pay_req))
-            .await
-            .map_err(|_| Error::PaymentFailed)?
-            .into_inner();
-
-        let total_amount = payment_response
-            .payment_route
-            .map_or(0, |route| route.total_amt_msat / MSAT_IN_SAT as i64)
-            as u64;
+        // Detect partial payments
+        if let Some(part_amt) = partial_amount {
```

Comment:
NIT: Since we care about both paths the `Some` and `None` we should use a `match` and not `if let` to be consistent with code style https://github.com/cashubtc/cdk/blob/main/CODE_STYLE.md#if-let